### PR TITLE
Tank Prefab

### LINKED ISSUE #14: User controlled tank
- [x] WASD for movement
- [x] Mouse for aiming and shooting
- [x] Unlimited bullets

### PR DESCRIPTION
**Pull request by djhuistra**
Closes #14 

- WASD for movement
- Mouse for aiming and shooting
- Unlimited bullets

image

### CODE REVIEW

File: Assets/TankController.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+					DestroyObject (goBullet);
+					projectiles.Remove (goBullet);
+				}
+			}
+		}
+	}
+
+
+
+//	void OnTriggerEnter2D(Collider2D other) {
```
Comment (ratenbuuren) [inline Assets/TankController.cs]: Remove

**Comment by ratenbuuren** [comment]
Add player tag to the player prefab please

**Comment by ratenbuuren** [comment]
Sorting layer unknown for tank prefab

**Comment by djhuistra** [comment]
New commits adds Tag, Layer and Sorting Layer to Tank and Bullet Prefabs. Also adds new movement system and removes commented code

File: Assets/Scripts/TankController.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+			mousePos = new Vector3(mousePos.x, mousePos.y, 0);
+
+			//Rotate the sprite to the mouse point
+			Vector3 diff = mousePos - bullet.transform.position;
+			diff.Normalize();
+			float rot_z = Mathf.Atan2(diff.y, diff.x) * Mathf.Rad2Deg;
+			bullet.transform.rotation = Quaternion.Euler(0f, 0f, rot_z - 90);
+
+			projectiles.Add (bullet);
+
```
Comment (ratenbuuren) [inline Assets/Scripts/TankController.cs]: empty line

File: Assets/Scripts/TankController.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		}
+
+		for (int i = 0; i < projectiles.Count; i++) {
+
+			GameObject goBullet = projectiles [i];
+			if (goBullet != null) {
+
+				//Move the sprite towards the mouse
+				goBullet.transform.position += goBullet.transform.up * projectileVelocity * Time.deltaTime;
+
```
Comment (ratenbuuren) [inline Assets/Scripts/TankController.cs]: empty line (2)

File: Assets/Scripts/TankController.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+		}
+
+		for (int i = 0; i < projectiles.Count; i++) {
+
+			GameObject goBullet = projectiles [i];
+			if (goBullet != null) {
+
+				//Move the sprite towards the mouse
+				goBullet.transform.position += goBullet.transform.up * projectileVelocity * Time.deltaTime;
```
Comment (ratenbuuren) [inline Assets/Scripts/TankController.cs]: Add script to Bullet prefab that moves it. Not the responsibility of tank.

File: Assets/Scripts/TankController.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+
+				// Destory Bullets if they get outside the screen
```
Comment (ratenbuuren) [inline Assets/Scripts/TankController.cs]: Less in update = better. 
Add to new bullet script:

private float timeToLive = 5f;

    void Start()
    {
        Destroy(this.gameObject, timeToLive);
    }

**Review by ratenbuuren — CHANGES_REQUESTED** [review]
Small improvements. Works as intended for MVP 👍

**Review by ratenbuuren — APPROVED** [review]

**Pull request merged**